src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -9,6 +9,174 @@ namespace Microsoft.ML.Runtime.Internal.CpuMath
+            int posMin, int iposMin, int iposLim, AlignedArray dst, int crun)
+        {
+            Contracts.AssertValue(rgposSrc);
+            Contracts.Assert(0 <= iposMin && iposMin <= iposLim && iposLim <= rgposSrc.Length);
+            Contracts.Assert(mat.Size == dst.Size * srcValues.Size);
+
+            if (iposMin >= iposLim)
+            {
+                if (!add)
+                    dst.ZeroItems();
+                return;
+            }
+
+            Contracts.AssertNonEmpty(rgposSrc);
+
+            if (Sse.IsSupported)
+            {
+                if (!tran)
+                {
+                    Contracts.Assert(0 <= crun && crun <= dst.Size);
+                    SseIntrinsics.MatMulPA(add, mat, rgposSrc, srcValues, posMin, iposMin, iposLim, dst, crun, srcValues.Size);
+                }
+                else
+                {
+                    Contracts.Assert(0 <= crun && crun <= srcValues.Size);
+                    SseIntrinsics.MatMulTranPA(add, mat, rgposSrc, srcValues, posMin, iposMin, iposLim, dst, dst.Size);
+                }
+            }
+            else
+            {
+                if (!tran)
+                {
+                    Contracts.Assert(0 <= crun && crun <= dst.Size);
+                    for (int i = 0; i < crun; i++)
+                    {
+                        float dotProduct = 0;
+                        for (int j = iposMin; j < iposLim; j++)
+                        {
+                            int col = rgposSrc[j] - posMin;
+                            dotProduct += mat[i * srcValues.Size + col] * srcValues[col];
+                        }
+
+                        if (add)
+                        {
+                            dst[i] += dotProduct;
+                        }
+                        else
+                        {
+                            dst[i] = dotProduct;
+                        }
+                    }
+                }
+                else
+                {
+                    Contracts.Assert(0 <= crun && crun <= srcValues.Size);
+                    for (int i = 0; i < dst.Size; i++)
+                    {
+                        float dotProduct = 0;
+                        for (int j = iposMin; j < iposLim; j++)
+                        {
+                            int col = rgposSrc[j] - posMin;
+                            dotProduct += mat[col * dst.Size + i] * srcValues[col];
+                        }
+
+                        if (add)
+                        {
+                            dst[i] += dotProduct;
+                        }
+                        else
+                        {
+                            dst[i] = dotProduct;
+                        }
+                    }
+
+                }
+            }
+        }
+
+        public static void Add(float a, float[] dst, int count)
+        {
+            Contracts.AssertNonEmpty(dst);
+            Contracts.Assert(0 < count);
```

Comment:
`count > 0`? I know it is just the order, but we usually read from left to right, so if you change the sides it is read differently. (0 is smaller than count -> count is greater than 0)